### PR TITLE
ADEN-2212 Hop in floor ad

### PR DESCRIPTION
**Pull request by gabrys**

### CODE REVIEW

File: extensions/wikia/AdEngine/js/provider/gpt/adDetect.js [hunk 1]
```diff
@@ -178,21 +180,21 @@ define('ext.wikia.adEngine.provider.gpt.adDetect', [
...
+			log(['msgCallback', slotName, 'caught message', data], 'info', logGroup);
 
 			if (data.status === 'success') {
-				if (expectAsyncSuccess) {
+				if (expectAsyncSuccess || expectAsyncHopWithSlotName) {
 					callAdCallback(data.extra);
 				} else {
 					log(
-						['msgCallback', slotname, 'Got asynchronous success message, while not expecting it'],
+						['msgCallback', slotName, 'Got asynchronous success message, while not expecting it'],
```
Comment (nandy-andy) [inline extensions/wikia/AdEngine/js/provider/gpt/adDetect.js:201]: Does it matter if it's a success or hop in this log/error message?

File: extensions/wikia/AdEngine/js/template/floor.js [hunk 2]
```diff
@@ -30,11 +31,17 @@ define('ext.wikia.adEngine.template.floor', [
 	 * Note the standard event bubbling applies, so it's possible some element within the iframe
 	 * stops the event propagation. Flash will stop propagation for sure.
 	 *
-	 * @param {Object} params
-	 * @param {string} params.code HTML code to put into floor container
-	 * @param {number} params.width width of the ad to put into floor container
-	 * @param {number} params.height width of the ad to put into floor container
-	 * @param {number} [params.onClick] function to call when floor iframe is clicked
+	 * If you supply detectAd, the code sniffs the ad in the supplied code by inspecting the iframe
```
Comment (nandy-andy) [inline extensions/wikia/AdEngine/js/template/floor.js]: `detectAd` or `canHop`?

File: extensions/wikia/AdEngine/js/template/floor.js [hunk 3]
```diff
@@ -30,11 +31,17 @@ define('ext.wikia.adEngine.template.floor', [
...
+	 * contents. The code will issue postMessage with either success or hop, so you need to pass
+	 * the params.slotName as well. The creative will need to have AdEngine_adType = 'floor'
+	 *
+	 * @param {Object}  params
+	 * @param {string}  params.code HTML code to put into floor container
+	 * @param {number}  params.width width of the ad to put into floor container
+	 * @param {number}  params.height width of the ad to put into floor container
+	 * @param {number}  [params.onClick] function to call when floor iframe is clicked
+	 * @param {boolean} [params.canHop] detect ad in the embedded iframe
+	 * @param {string}  [params.slotName] name of the original slot (required if detectAd)
```
Comment (nandy-andy) [inline extensions/wikia/AdEngine/js/template/floor.js]: shouldn't it reads: required if `canHop` set to `true`?

File: extensions/wikia/AdEngine/js/provider/gpt/adDetect.js [hunk 4]
```diff
@@ -149,14 +149,16 @@ define('ext.wikia.adEngine.provider.gpt.adDetect', [
...
-		var adType = forcedAdType || getAdType(slotname, gptEvent, iframe),
+		var adType = forcedAdType || getAdType(slotName, gptEvent, iframe),
 			shouldPollForSuccess = false,
 			expectAsyncHop = false,
 			expectAsyncHopWithSlotName = false,
+			expectAsyncSuccessWithSlotName = false,
 			expectAsyncSuccess = false,
-			successTimer;
+			successTimer,
+			shortSlotName = slotName.replace(/^.*\/([^\/]*)$/, '$1');
```
Comment (nandy-andy) [inline extensions/wikia/AdEngine/js/provider/gpt/adDetect.js]: Everytime I see a regex I think: "Unit test!" ;)

Unit test all the things!

File: extensions/wikia/AdEngine/js/provider/gpt/adDetect.js [hunk 5]
```diff
@@ -178,21 +184,21 @@ define('ext.wikia.adEngine.provider.gpt.adDetect', [
...
 			}, 500);
 		}
 
 		function msgCallback(data) {
-			log(['msgCallback', slotname, 'caught message', data], 'info', logGroup);
+			log(['msgCallback', slotName, 'caught message', data], 'info', logGroup);
 
 			if (data.status === 'success') {
-				if (expectAsyncSuccess) {
+				if (expectAsyncSuccess || expectAsyncHopWithSlotName) {
```
Comment (fraszczakszymon) [inline extensions/wikia/AdEngine/js/provider/gpt/adDetect.js]: Is here **expectAsyncHopWithSlotName** variable used correctly? Shouldn't be here **expectAsyncSuccessWithSlotName**? If it is correct it seems that expectAsyncSuccessWithSlotName is reduntant to expectAsyncHopWithSlotName

Comment (gabrys) [inline extensions/wikia/AdEngine/js/provider/gpt/adDetect.js]: Yes, should be expectAsyncSuccessWithSlotName.

**Comment by fraszczakszymon** [comment]
LGTM :+1:

**Pull request merged**